dvc/remote/base.py

```diff
@@ -165,7 +165,7 @@ def changed(self, path_info, checksum_info):
         return False
 
     def save(self, path_info):
-        raise NotImplementedError
+        raise NotImplementedError("save not supported by this remote")
```

Comment:
I think this is good enough. Just move it into a new class please and raise it instead of copying this string over and over again :)

Reply:
Btw, you could use `self.scheme` to tailor messages even more :) E.g. something like `raise NotImplementedError("{} is not supported by {} remote".format("save", self.scheme))`.